cpp/oneapi/dal/algo/jaccard/backend/cpu/vertex_similarity_default_kernel.cpp

```diff
@@ -514,18 +548,21 @@ template size_t intersection<uint64_t>(uint64_t *neigh_u,
                                        uint64_t n_u,
                                        uint64_t n_v);
 
+template <typename Graph>
 similarity_result call_jaccard_block_kernel(const descriptor_base &desc,
-                                            const similarity_input &input) {
+                                            const similarity_input<Graph> &input) {
     std::cout << "Jaccard block kernel started" << std::endl;
+
     const auto my_graph = input.get_graph();
-    std::cout << get_vertex_count(my_graph) << std::endl;
-    std::cout << get_edge_count(my_graph) << std::endl;
+
+    std::cout << get_vertex_count_impl(my_graph) << std::endl;
+    std::cout << get_edge_count_impl(my_graph) << std::endl;
     auto node_id = 0;
-    std::cout << "degree of " << node_id << ": " << get_vertex_degree(my_graph, node_id)
+    std::cout << "degree of " << node_id << ": " << get_vertex_degree_impl(my_graph, node_id)
               << std::endl;
-    for (unsigned int j = 0; j < get_vertex_count(my_graph); ++j) {
+    for (unsigned int j = 0; j < get_vertex_count_impl(my_graph); ++j) {
```

Comment:
is this a call to the function? Does the compiler optimize it?

Reply:
Yes, it's a call.